artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ChannelImpl.java

```diff
@@ -151,6 +151,11 @@ public ChannelImpl(final CoreRemotingConnection connection,
       }
 
       this.interceptors = interceptors;
+      //zero copy transfer is initialized only for replication channels
```

Comment:
@clebertsuconic This one is ugly I know :) 
We can find a better way together ;)

Reply:
Yeah.. this looks like a Hack to me! when is sendFiles used without being the replica channel?

Reply:
Right now is not and I've added an assert to make the test suite to fail on it, but we can always configure the Netty pipeline to handle them with no harm, If is more clear..